doc: Improve gui/src/qt README.md

jarolrod opened the pull request:
**Master/Before:** [Render of Master](https://github.com/bitcoin-core/gui/blob/master/src/qt/README.md)


**PR/After:** [Render of PR](https://github.com/bitcoin-core/gui/blob/5d1f260713f9f1d29c0db68f2917dfe7368d4ba0/src/qt/README.md)

**Changes:**
The README.md found in `gui/src/qt` seems to not have gotten any love in a while. This PR fixes some grammatical errors, makes it easier to follow, and modernizes the logic of using Qt Creator. 
1. Makes several sections more informative
2. Directories under `Files and Directories` now end with a forward slash denoting that they are a directory
3. Modernize the Qt Creator Logic for the current setup flow
4. Add UNIX Qt Creator Setup Instructions (Ubuntu & Debian)

hebasto reviewed (commented):
For editing `*.ui` files the `Qt Designer` is enough for me, but this opinion is personal, ofc.

jarolrod commented on src/qt/README.md:
```diff
@@ -1,62 +1,61 @@
-This directory contains the BitcoinQT graphical user interface (GUI). It uses the cross-platform framework [Qt](https://www1.qt.io/developers/).
+This directory contains the source code for the reference implementation of the Bitcoin Core graphical user interface (GUI). It uses the [Qt](https://www1.qt.io/developers/) cross-platform framework.
```
I was drawing on conversations had about the role of the Bitcoin Core GUI. Here is an example https://github.com/bitcoin-core/gui/issues/45#issuecomment-671333460. I think it's OK to call this the reference implementation. If it's not OK with the community, I'll change this to: `This directory contains the source code for the Bitcoin Core graphical user interface (GUI).`

RandyMcMillan replied:
I like this better..

`
This directory contains the source code for the Bitcoin Core graphical user interface (GUI).
`

It is more correct IMO.

jarolrod replied:
Addressed this in f6584d9c84f4d840dd5cb49f7dcc9984e9a16304. Used @RandyMcMillan suggestion, thanks!

jarolrod commented on src/qt/README.md:
```diff
@@ -1,62 +1,61 @@
+## Files and Directories
 
-### forms
+#### /forms
```
You're right, `forms/` is more appropriate

jarolrod commented on src/qt/README.md:
```diff
@@ -1,62 +1,61 @@
+#### /locale
 
-Contains translations. They are periodically updated. The process is described [here](/doc/translation_process.md).
+- Contains translations of the strings used throughout the GUI. They are periodically updated and an effort is made to support as many languages as possible. The process of contributing translations is described [here](/doc/translation_process.md).
```
gui/src/qt/locale/ seems to only contain translations for the GUI

hebasto replied:
https://github.com/bitcoin-core/gui/blob/7ae86b3c6845873ca96650fc69beb4ae5285c801/src/qt/locale/bitcoin_en.ts#L4547

https://github.com/bitcoin-core/gui/blob/7ae86b3c6845873ca96650fc69beb4ae5285c801/src/init.cpp#L1127

jarolrod replied:
addressed this in f6584d9c84f4d840dd5cb49f7dcc9984e9a16304. Removed the `gui` specific part.

jonasschnelli commented on src/qt/README.md:
```diff
@@ -1,62 +1,61 @@
+#### paymentserver.(h/cpp)
 
-### paymentserver.(h/cpp)
+- Used to process BIP21 payment URI requests. Also handles URI based application switching (e.g. when following a bitcoin:... link from a browser).
```
Maybe add `(deprecated)`?

jarolrod replied:
Done in f6584d9c84f4d840dd5cb49f7dcc9984e9a16304

jonasschnelli commented on src/qt/README.md:
```diff
@@ -75,21 +74,23 @@ See [CONTRIBUTING.md](/CONTRIBUTING.md) for general guidelines. Specifically for
+8. In the `Projects` tab, select `Manage Kits...`
+ - Under `Kits`: select the default "Desktop" kit
+ - Under `Compilers`: select an appropriate compiler. (on macOS select `"Clang (x86 64bit in /usr/bin)"`)
+ - Under `Debuggers`: select `"LLDB"` as debugger (you might need to set the path to your LLDB installation)
```
I guess this is Mac specific (clang in /usr/bin/ as well as `LLDB`). I would probably keep this as "Mac" instructions (Linux/Win) people might need to adapt but they are not blinded by not telling them its made for Mac.

jarolrod replied:
Done in f6584d9c84f4d840dd5cb49f7dcc9984e9a16304

jonasschnelli commented:
ACK modulo 2xcomments

RandyMcMillan commented:
@jarolrod - You should be able to trigger a rebuild yourself.
https://cirrus-ci.com/task/6120245750398976

ACK :)

RandyMcMillan commented:
The parent commits never passed CI

The current parent commits

https://github.com/bitcoin-core/gui/commit/ca759bc743c9b3b240289c467349cefc5f4444af
https://github.com/bitcoin-core/gui/commit/62c93c34c1d9b83cd168b6b1e30d10e01db53e38
https://github.com/bitcoin-core/gui/commit/b40d4d494e1e87fecdba05463430749971c1b023

I suggest rebasing it to 

https://github.com/bitcoin-core/gui/commit/a0489f3472f3799dc1ece32a59556fd239c4c14b

As you can see - it passed CIRRUS

https://cirrus-ci.com/build/4908623950249984

and travis-ci

https://travis-ci.org/github/bitcoin-core/gui/builds/747420647

jarolrod commented on src/qt/README.md:
```diff
@@ -1,62 +1,61 @@
 ## Compile and run
 
-See build instructions ([macOS](/doc/build-osx.md), [Windows](/doc/build-windows.md), [Unix](/doc/build-unix.md), etc).
+See build instructions ([macOS](/doc/build-osx.md), [Windows](/doc/build-windows.md), [Unix](/doc/build-unix.md), etc). Compiling generates the `bitcoin-qt` executable.
```
Addresses this in f6584d9c84f4d840dd5cb49f7dcc9984e9a16304. Removed this string in favor of just referring users to the build instruction.

jarolrod commented:
updated to f6584d9c84f4d840dd5cb49f7dcc9984e9a16304

Addressed @hebasto [comments](https://github.com/bitcoin-core/gui/pull/139#pullrequestreview-540386515) and @jonasschnelli [suggestions](https://github.com/bitcoin-core/gui/pull/139#issuecomment-736355489)

jonatack commented on src/qt/README.md:
```diff
@@ -1,62 +1,61 @@
+#### forms/
 
-Contains [Designer UI](https://doc.qt.io/qt-5.9/designer-using-a-ui-file.html) files. They are created with [Qt Creator](#using-qt-creator-as-ide), but can be edited using any text editor.
+- A directory which contains [Designer UI](https://doc.qt.io/qt-5.9/designer-using-a-ui-file.html) files. These files specify the characteristics of form elements in XML. Qt UI files are normally altered with [Qt Creator](#using-qt-creator-as-ide), but can be edited using any text editor.
```
- s/which/that/
- s/normally altered/can be edited/
- s/, but can be edited/or/

jonatack reviewed (commented):
Concept ACK

jarolrod commented:
updated f6584d9 -> a9e641d

Implemented @jonatack suggestions. Thanks for the in-depth review!

Additionally, I have included some Qt Creator setup instructions for UNIX (Ubuntu & Debian). Windows instructions still need to be done. I will take care of that in a follow-up PR.

New Render: [src/qt/README.md](https://github.com/bitcoin-core/gui/blob/a9e641d40d83f1a89bc57ab14f39ed370e0c7e2a/src/qt/README.md)

jonatack commented:
Thanks, will re-review.

jonatack commented on src/qt/README.md:
```diff
@@ -69,27 +70,54 @@ Represents the view to a single wallet.
+ **macOS**
+ - Under `Kits`: select the default "Desktop" kit
+ - Under `Compilers`: select `"Clang (x86 64bit in /usr/bin)"`
+ - Under `Debuggers`: select `"LLDB"` as debugger (you might need to set the path to your LLDB installation)  
```
remove whitespace at EOL
```diff
  - Under `Compilers`: select `"Clang (x86 64bit in /usr/bin)"`
- - Under `Debuggers`: select `"LLDB"` as debugger (you might need to set the path to your LLDB installation)  
+ - Under `Debuggers`: select `"LLDB"` as debugger (you might need to set the path to your LLDB installation)
 
- **Ubuntu & Debian**  
+ **Ubuntu & Debian**
  **Note:** Some of these options may already be set
```

jonatack commented on src/qt/README.md:
```diff
@@ -1,62 +1,63 @@
+- Represents the view to a single wallet.
 
-### Other .h/cpp files
+#### Other .h/cpp files
```
lines 64 and 66: s/etc/etc./

jonatack reviewed (commented):
Almost-ACK, just a couple more items below.

jarolrod commented:
updated a9e641d -> 63d0f29c9945dcf512bf494a4f24190121c48648, addressed @jonatack [suggestions](https://github.com/bitcoin-core/gui/pull/139#pullrequestreview-573287949)

jonatack commented on src/qt/README.md:
```diff
@@ -1,95 +1,123 @@
 
-Download and install the community edition of [Qt Creator](https://www.qt.io/download/).
-Uncheck everything except Qt Creator during the installation process.
+ **Ubuntu & Debian**  
```
```suggestion
 **Ubuntu & Debian**
```

jarolrod replied:
addressed in 5d1f260

jarolrod commented:
updated 63d0f29c9945dcf512bf494a4f24190121c48648 -> 5d1f260, addressed leading whitespace issue

jonatack reviewed (approved):
ACK 5d1f260713f9f1d29c0db68f2917dfe7368d4ba0

RandyMcMillan commented:
ACK 👍🏼
